runtime/src/bank.rs

```diff
@@ -752,6 +758,29 @@ impl Bank {
 
     /// Create a new bank that points to an immutable checkpoint of another bank.
     pub fn new_from_parent(parent: &Arc<Bank>, collector_id: &Pubkey, slot: Slot) -> Self {
+        Self::_new_from_parent(
+            parent,
+            collector_id,
+            slot,
+            &mut None::<fn(&RewardCalcEvent)>,
+        )
+    }
+
+    pub fn new_from_parent_with_tracer(
+        parent: &Arc<Bank>,
+        collector_id: &Pubkey,
+        slot: Slot,
+        tracer: &mut Option<impl FnMut(&RewardCalcEvent)>,
```

Comment:
I wanted to factor this ugly type signature out; but I couldn't mainly the limitation of `impl Trait`.